src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,59 @@
+package duke.task;
+
+import duke.DukeException;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public void markTask(int taskId, boolean isDone) throws DukeException {
+        try {
+            if (isDone) {
+                tasks.get(taskId).markAsDone();
+            } else {
+                tasks.get(taskId).markAsNotDone();
+            }
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException(e.getMessage());
+        }
+    }
+
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    public TaskList filterTask(String keywords) {
```

Comment:
Perhaps a function name which sounds more like a plural as the function is filtering among many tasks?

```suggestion
    public TaskList filterTasks(String keywords) {
```